src/chromeDebugAdapter.ts

```diff
@@ -433,6 +431,45 @@ export class ChromeDebugAdapter extends CoreDebugAdapter {
         }
     }
 
+    private async spawnChromeUnelevatedWithWindowsScriptHost(chromePath: string, chromeArgs: string[]): Promise<number> {
+        const semaphoreFile = path.join(os.tmpdir(), 'launchedUnelevatedChromeProcess.id');
+        if (fs.existsSync(semaphoreFile)) { // remove the previous semaphoreFile if it exists.
+            fs.unlinkSync(semaphoreFile);
+        }
+        const chromeProc = fork(getChromeSpawnHelperPath(),
+            [`${process.env.windir}\\System32\\cscript.exe`, path.join(__dirname, 'launchUnelevated.js'),
+            semaphoreFile, chromePath, ...chromeArgs], {});
+
+        chromeProc.unref();
+        await new Promise<void>((resolve, reject) => {
+            chromeProc.on('message', resolve);
+        });
+
+        const pidStr = await findNewlyLaunchedChromeProcess(semaphoreFile);
+
+        if (pidStr) {
+            logger.log(`Parsed output file and got Chrome PID ${pidStr}`);
+            return parseInt(pidStr, 10);
+        }
+
+        return null
+    }
+
+    private async spawnChromeUnelevatedWithPineZorro(chromePath: string, chromeArgs: string[]): Promise<number> {
+        return new Promise<number>((resolve, reject) => {
+            this._session.sendRequest("launchUnelevated", {
```

Comment:
We are wrapping sendRequest in a promise in a similar way here: https://github.com/Microsoft/vscode-chrome-debug-core/blob/47782a37b6df5efd3a50bf1813019c0068a984fe/src/transformers/fallbackToClientPathTransformer.ts#L36
Would it make sense to extract the async sendRequest logic someplace else?

Reply:
Does it need telemetry?

Reply:
@digeff The two usages are not quite the same, they are very different in the parameters they each have respectively. Having a common utility function will carry the different parameters as well, it almost like repeating this code. So extracting a common logic does not improve much, but reduces readability.

@roblourens if the operation does not succeed it will reject, and the call will get an exception thrown then telemetry handler will capture that. Otherwise, there is C# telemetry implemented in the host for each PZ extension requests.

Reply:
@dhanvikapila Yes, the admin flag is right on the same DA event (clientRequest/launch). The C# error is on a separate event.